### PR TITLE
Enhance Find by

### LINKED ISSUE #91: Make some optimised findBy implementations

### PR DESCRIPTION
**Pull request by GTZL1**
I still think we can factorise them more (and IDE tells it too), but we'll see about this later.
Fix #91

### CODE REVIEW

File: src/main/java/ch/heigvd/amt/stack/infrastructure/persistence/database/JdbcQuestionRepository.java [hunk 1]
```diff
@@ -34,21 +34,30 @@ private DataSource getDataSource() {
...
-                                    .contains(query.getShouldContain().toLowerCase())
-                    ) || (
-                            question.getDescription() != null && question.getDescription()
-                                    .toLowerCase()
-                                    .contains(query.getShouldContain().toLowerCase())
-                    ))
-                    .collect(Collectors.toList());
-        } else {
-            return findAll();
+        var select = "SELECT * FROM Question WHERE title = ? OR description = ?;";
```
Comment (alexandrepiveteau) [inline src/main/java/ch/heigvd/amt/stack/infrastructure/persistence/database/JdbcQuestionRepository.java:37]: We are not looking for exact matches, but rather contents that contain the query with arbitrary case.

File: src/main/java/ch/heigvd/amt/stack/infrastructure/persistence/database/JdbcRepository.java [hunk 2]
```diff
@@ -27,7 +27,7 @@
             "CREATE TABLE IF NOT EXISTS Session" +
                     "( idSession VARCHAR PRIMARY KEY" +
                     ", idxCredential VARCHAR" +
-                    ", tag VARCHAR(50)" +
+                    ", tag VARCHAR(50) PRIMARY KEY " +
```
Comment (alexandrepiveteau) [inline src/main/java/ch/heigvd/amt/stack/infrastructure/persistence/database/JdbcRepository.java:30]: Why would we need this ?

**Review by alexandrepiveteau — CHANGES_REQUESTED** [review]

File: src/main/java/ch/heigvd/amt/stack/infrastructure/persistence/database/JdbcRepository.java (hunk 2 above)
Comment (GTZL1) [reply src/main/java/ch/heigvd/amt/stack/infrastructure/persistence/database/JdbcRepository.java:30]: We didn't agreed yesterday that tag is also part of the PK ?

Comment (alexandrepiveteau) [reply src/main/java/ch/heigvd/amt/stack/infrastructure/persistence/database/JdbcRepository.java:30]: No 🙂

**Pull request closed without merging**